### PR TITLE
WIP: Rego #957 adds profiler scripts for regolith and gooey debugging

### LINKED ISSUE #957: add profiling versions of the scripts for dev debugging
we currently have `regolith` and `regolith..bat` and likewise for the helper gui as scripts.

make the same for `profile_regolith` and `profile_regolith.bat` and likewise for the helper gui.

These are not user-facing but it would be good to have them permanently there for the use of devs.  No need to update the documentation and no need for any tests, but we will need a news item for the changelog.

The insides will look a bit like this;
```
#!/usr/bin/env python
import cProfile

from regolith.main import main

cProfile.run("main()", sort="cumtime")
```

### PR DESCRIPTION
**Pull request by dpeters001**
Closes #957 
Adds profiler version of regolith and helper_gui for devs. Setup.py is modified to add these scripts to the conda environment in which regolith is installed. ```profile_regolith helper``` and ```profile_helper_gui``` can be used by devs to create a log of function calls associated with queries.

### CODE REVIEW

**Comment by dpeters001** [comment]
> looks good. need line feeds at the ends of the files and a news and i can merge

Added the news file. Let me know what you think. I want to make sure that I am communicating the additions clearly.

File: news/add_profiling_versions_of_scripts.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+**Added:**
+ * profile_regolith for a profiling version of the regolith command to be used for developer debugging and the creation of function call logs
```
Comment (sbillinge) [inline news/add_profiling_versions_of_scripts.rst:2]: Great job.  I think I would combine these though. Something like
```
* scripts (profile_regolith and profile_helper_gui) that run regolith and regolith_helper with the python cProfiler.  Used for debugging and code development.
```

File: news/add_profiling_versions_of_scripts.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * profile_helper_gui for a profiling version of the helper_gui command to be used for developer debugging and the creation of function call logs
+
+**Changed:**
+ * setup.py to install profiling commands to a new conda environment during regolith installation
```
Comment (sbillinge) [inline news/add_profiling_versions_of_scripts.rst:6]: This is not needed because it is part of the code update described above.

**Review by sbillinge — COMMENTED** [review]
This looks good.  For some reason it is failling a bunch of tests.  This also happened to Kayla.  We are not sure what happened.  It could be some change that happened automatically when the version of some package changed.  It is because a bunch of the output files have or have missing a line feed at the end.   Please try and update those files to make the tests pass.  I couldn't reproduce that behavior when I updated my master and ran the tests locally so not sure what it going on, but if you click on `details` in the failing CI on GitHub and click through to Travis, you will see this.  Open the test output file for the scopatzI think it is the last line of the scopatz.bib or sbillinge.bib file or sthg.  Maybe work with Nick if you need to.  It is a puzzle but we have to try and figure out what is going on.

**Comment by sbillinge** [comment]
@dpeters001 please can you make the edit to the news and I can merge this.   Even better, if you can click the "allow maintainer to edit PRs" button (not sure exactly what it is called) it will allow me to make small edits directly on the PRs in the future.

**Comment by dpeters001** [comment]
- Committed the news file edits. 
- I have the "allow edits by maintainers" box checked for this PR and will make sure it is checked for future PRs.

**Comment by sbillinge** [comment]
Thanks @dpeters001

**Pull request merged**